### ISSUE TITLE
Word Wrap does not apply to Find results

### ISSUE THREAD

**Issue by Leif-W** [notepad-plus-plus/notepad-plus-plus#1811]
Currently, with Word Wrap selected, only the file contents are word wrapped.  However, if we do a Find All (in Current or in All Opened) Document(s), the "Find result" window or pane has no word wrap.  This is a usability problem, as it results in long lines requiring scrolling in 2 directions instead of 1 (not all mouse wheels have a "tilt" function to allow 2-axis scrolling), and an entire line can not be easily read at a glance.  If a user specifies that they want Word Wrap to display the file, then be consistent and wrap the lines of that file, regardless what context they appear.

**Comment by MetaChuh** [comment]
this behaviour is by design, to keep the vertical size as low as possible.

thank you for your help and best regards.

-----
```
general notes:

please close your issues as soon as you are satisfied, to help
keeping the notepad++ issue tracker clean.

when creating a new issue, please provide all information as shown at the issue template.
your notepad++ debug info is part of the minimum requirements.
(? > Debug Info... > Copy debug info into clipboard).

please also conduct a search at the issue tracker beforehand, to check if a similar issue
already exists.

for general questions, or if it is not sure, whether your issue is directly related to
the notepad++ source code, please visit us at the notepad++ community forum, and
search if related topics exists. you are welcome to post either at similar
topics, or to create a new topic at Help Wanted or General Discussion.
(no extra account is needed, just use your github account to sign in)
```
[<p align="center"><kbd> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; Click here to visit the Notepad++ Community Forum &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;&emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; </kbd></p>](https://notepad-plus-plus.org/community/)

**Comment by Leif-W** [comment]
Understand that it was by design, but also it is a poor design choice.  The choice of such behavior should be left to the user.  It is in many cases preferable to see at least 2-3 lines wrapped, as vertical space of an undocked _Find Results_ window is plentiful.  It is a premature optimization based upon the potentially false assumption that vertical space is a primary limiting factor.  However, productivity - a user's time and energy - may be a much more scare resource and thus a higher priority for optimization.  Much more time  is saved by seeing all info at a glance and scrolling in a single vertical axis with a mouse wheel rather than trying to scroll on two axis, by repositioning the mouse, dragging left or right, releasing the drag to scroll up and relize you must scroll back to view the contents of a previous match for quick comparison.  This compound loss of productivity **BY FAR** outweighs and invalidates almost all concern in almost all cases, of vertical real-estate optimization.  Therefore this optimization is premature and based upon false assumptions and illogical reasoning.  This is basic UX design theory here.  However, it may be valid in some edge cases where a person may prefer this behavior, and so it should be available as a non-default option.

**Comment by sasumner** [comment]
I managed to script this behavior; not sure if I like the result or not:

![image](https://user-images.githubusercontent.com/30118311/56873269-2d5a9980-69ff-11e9-8aeb-5d4791a8c6f6.png)

**Comment by sba923** [comment]
> I managed to script this behavior; not sure if I like the result or not:

how did you achieve that?

**Comment by sasumner** [comment]
@sba923 Well, it was a bit complicated to (Python)script it; the hardest part being getting a "handle" to the Find result window, which is just another Scintilla window like the main editor window(s), but it has an additional complication that it doesn't exist before you conduct a search in Notepad++.  Once you have the handle you can call Scintilla functions on it using the IDs in the Scintilla.iface file.  So that's what I did using ID 2268 to "SetWrapMode".

**Comment by sasumner** [comment]
This feature is in Notepad++ as of 7.9.
See https://github.com/notepad-plus-plus/notepad-plus-plus/commit/3a2edb99c40f5577a29307f1321d9a3cbe2a69cb
Some quickie instructions for usage are https://github.com/notepad-plus-plus/notepad-plus-plus/pull/8681#issuecomment-670977364

**Comment by MichelMerlin** [comment]
**How do I set Word wrap for Search results pane?**
My Search results pane has had Word wrap for years, but yesterday I lost it (pane disappeared, so, using one of the advices in [[Notepad++ Search Window disappeared](https://stackoverflow.com/questions/46632070/notepad-search-window-disappeared)), I went to "C:\Users\...\AppData\Roaming\Notepad++" and renamed "config.xml" as "config.xml.BAD", which did retrieved the SR pane but of course also lost a lot of my personal settings. Hard to restore them from config.xml.BAD because too many settings have changed and most have obscur names.
In those numerous lost settings is the Search results pane Word wrap. So I badly need to know how to set it. TIA to any help.
Versailles, Sun 18 Sep 2022 23:46:45 +0200

**Comment by alankilborn** [comment]
@MichelMerlin Right above your post is a "quickie instructions for usage" link...

**Comment by MichelMerlin** [comment]
"[alankilborn](https://github.com/alankilborn), Thanks for your timely reply, but above [your post of 10h59 GMT](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/1811#issuecomment-1250404845) is my post that you mentioned where, by precaution against the usual useless rants or comments, I used enough place to mention that I did read and use the previous threads about that question, including of course [the one you now re-refer](https://github.com/notepad-plus-plus/notepad-plus-plus/pull/8681#issuecomment-670977364). Should I spend a couple pages and the according time to report it all?
Instead I suggest that, if you know how to _answer_ (not glose about) my question (« **How do I set Word wrap for Search results pane?** »), you either explain how to do it (_with the required precision and concision of course, i.e. telling precisely what is asked, with all the words needed and not a single more, as just repeating vague incantations as common on forums would be just waste of time and space_), or post an URL that does it (_I again mean with the required clarity hence precision yet concison_), then please do make a new post giving that info, and nothing else (_which would be irrelevant_).
Sorry for being long, but being short proves to only get the question and its author taken full mistakenly by the Fast Posters sort of owning the forum.
Versailles, Mon 19 Sep 2022 18:49:15 +0200

**Comment by alankilborn** [comment]
@MichelMerlin 

Actually, I suppose I must have misunderstood your question -- and the rephrasing (to something much longer) didn't clarify for me to be able to help you any further.

And truly, questions about the software belong [HERE](https://community.notepad-plus-plus.org/), not where we are now, so I'll let you take your question over there in seeking your answer.  

Or possibly your question is answered in the [USER MANUAL](https://npp-user-manual.org/) so you could also have a look there.

**Comment by MichelMerlin** [comment]
"[alankilborn](https://github.com/alankilborn)" Mon 19 sep 17h16 GMT « _misunderstood your question -- and the rephrasing... didn't clarify_ »
I don't see what in my twice repeated question « **How do I set Word wrap for Search results pane?** » can be "_misunderstood_".
If you know how, then please do make a post explaining, or better, pointing the very place where it is explained, in NP++'s community or manual, or anywhere else.
If you don't know, be smart enough to simply reply "_I don't know_", or even better, to NOT reply at all; as replying to just mention those 2 resources home pages (your 2 links), to someone familiar enough with NP++ to have reached the present much more specialized github page, makes you look like a newbie I-know-all of 1993 (Internet big bang).
It's your liberty to think you have more experience and knowledge of NP++ and life in general than me, it's mine to think the reverse. Let's continue each one his own path and opinions and keep respectful to the other and to all others.
What I retain is, NP++ certainly has a setting to turn on-off the _Search results pane Word wrap_ (since it appears on or off in various recent circumstances), but none so far knows where that setting is or how to turn it on or off.
Versailles, Tue 20 Sep 2022 16:07:10 +0200

**Comment by MichelMerlin** [comment]
**SOLVED - How do I set Word wrap for Search results pane?**

"[MichelMerlin](https://github.com/MichelMerlin)" Sun 18 Sep 2022 21:46:45 GMT « [**How do I set Word wrap for Search results pane?**](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/1811#issuecomment-1250392760) »
Self-answer:

Exit Notepad++
    In Windows Explorer Address bar, paste:
        %USERPROFILE%\AppData\Roaming\Notepad++
    or:
        %APPDATA%\Notepad++
    of which either will expand as:
        C:\Users\Audrey Hepburn\AppData\Roaming\Notepad++
    (replace Audrey Hepburn with your own Windows User Name)
    Then in that folder, find the file:
        config.xml
        and SAVE A COPY OF IT, naming it as e.g. "config.xml.2022-0920.162347.BAK"
        Note: Unspecified Time Zones should be seen as GMT
    WARNING: This backup is really necessary: Notepad++ configuration files often get entirely
        lost (contents replaced with irrelevant defaults) during mundane operations.
    Then IN ANOTHER TEXT EDITOR (e.g. Windows Notepad, PSpad, but NOT Notepad++), open it, then:
        set "wrappedLines" to "yes":
        <GUIConfig name="FinderConfig" wrappedLines="yes"...
        save config.xml, exit the alternative text editor
re-open Notepad++
Verify that config.xml, as it is now in Notepad++, does have « wrappedLines="yes" »
Verify that the Search results pane has now a regular word wrap.
Note: config.xml may be absent and appear only after a couple days and some changes in Notepad++ Settings and Preferences followed with exit-reopen Notepad++.

DETAILS AND EXPLANATIONS

DO NOT EDIT config.xml IN NOTEPAD++! This is because, in Notepad++, Settings and Preferences are:
    - supposed to be edited IN THE GUI (Graphical User Interface)
    - ONLY ON EXIT saved in the various configuration files
    - ON REBOOT copied from config.xml (or else) into the GUI
Details in https://npp-user-manual.org/docs/config-files
Note. That page suggests to edit config.xml in another instance of Notepad++, but this may cause confusion due to the remaining interferences between the various Notepad++ instances. I suggest instead to use a full different editor, e.g. PSpad ("PSPad.exe") or Windows Notepad ("%windir%\system32\notepad.exe").
Versailles, Wed 16 Nov 2022 17:12:10 +0100

**Comment by alankilborn** [comment]
@MichelMerlin 

Advising other users to do it that way is probably not the best idea.
Consult the appropriate place in the [USER MANUAL](https://npp-user-manual.org/docs/searching/#search-results-configuration-options) to see the proper way to do it.

**Comment by MichelMerlin** [comment]
Thanks [alankilborn](https://github.com/alankilborn) for giving a better solution. I appreciate it (The obvious often is the best and does need recalling😉)
Now giving it 2 months earlier instead of your post of Mon 19 Sep 2022 10:59 GMT (that didn't help me or others) would have been even more useful. Friendliness is more useful than [NIH attitude](https://en.wikipedia.org/wiki/Not_invented_here).
So let me (try) make your answer even clearer and more visible for others in need:

_Right-click inside the **Search result pane** then check **Word wrap long lines**_
Other details in <A href="https://npp-user-manual.org">Official Manual</A> > [Searching](https://npp-user-manual.org/docs/searching) > [Search results configuration options](https://npp-user-manual.org/docs/searching#search-results-configuration-options)

Now on _**How to edit Notepad++ configuration files**_ like config.xml,
- 1st read my [SOLVED - How do I set Word wrap for Search results pane?](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/1811#issuecomment-1317270331) of Wed 16 Nov 2022 16:12 GMT above. Don't be afraid by its length, a long complete precise clearly written text may be quicker to read than one may think.
- 2nd use the Reference, even longer and less clear yet more detailed and surely even more verified:
<A href="https://npp-user-manual.org">Notepad++ Manual</A> > [Configuration Files Details](https://npp-user-manual.org/docs/config-files) > [Editing Configuration Files](https://npp-user-manual.org/docs/config-files#editing-configuration-files)

Versailles, Wed 23 Nov 2022 09:51:45 +0100

**Comment by Leif-W** [comment]
When I opened this issue over 6 years ago, it was summarily dismissed and closed, with my feedback ignored, despite some thumbs up.  It should not have been closed, but left open and tagged as a feature request.  4 years later, @sasumner implemented a patch to add the feature, with a quick, simple, intuitive configuration of just right-clicking in the find results pane.  I wish to express gratitude to all who helped add this feature.  It makes life easier, and the user experience more enjoyable.

The initial [comment](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/1811#issuecomment-1251280926) from @MichelMerlin seems to be a separate problem.  The config files were corrupted, possibly at runtime, and Notepad++ was no longer able to create a Search Results window.  The subsequent verbose and complex comments seemed to describe the process of manually re-creating parts of the configuration file to recover from the error.  Did I understand correctly?   If yes, that probably all should have been addressed in it's own issue, possibly referencing this issue in passing.

Also, if I understood @MichelMerlin 's initial problem, then @alankilborn 's suggestion to just RTFM is inappropriate as it doesn't address the problem.  You can't right click on a search result window that doesn't exist because the program won't create the window, or if the window is created but the program doesn't create the word wrap option, either case possibly resulting from a corrupted config file.

@MichelMerlin wrote:

> So let me (try) make your answer even clearer and more visible for others in need:
> 
> Right-click inside the Search result pane then check Word wrap long lines

This was self-evident to me (and probably most people) from @sasumner 's "quickie instructions".  Restating it in text adds clarity.  Although redundant for most, clear communication in documentation is necessary to ensure comprehension.  Some understand with pictures, some with words, both are necessary and valid, neither is better or wrong.

As for config corruption, I've never experienced that.  I frequently experience Notepad++ just deleting the config entirely upon upgrade.  Possibly a separate installer issue.  Possibly related to my installation method via Chocolatey.  However, should I experience such a config corruption, then @MichelMerlin 's comments may prove insightful and helpful.  That is why I wish they were put in a separate issue, (or new issue if one doesn't already exist), so that others experiencing a config.xml file corruption could more easily find the issue (i.e. GitHub issue search, scanning by title in search results), and better understand the steps to recovery.  One might not intuitively guess or be reasonably expected to look under "Word Wrap does not apply to Find r..." for help on "Recover corrupt config.xml".

Anyways, thanks to all involved for the community, support, implementation, documentation, and recovery information.

**Comment by alankilborn** [comment]
@Leif-W 

It wasn't any sort of "corruption" problem.  Reading back thorough it now, it appears to just be someone that doesn't understand basic computer program usage.

>This was self-evident to me (and probably most people) from @sasumner 's "quickie instructions". 

Yes, in fact it was so basic, I had no idea what the "problem" was with the user not understanding, so it was not realistic to "restate it in text".

Of course, it seems obvious with hindsight.  :-P

**Comment by Leif-W** [comment]
> It wasn't any sort of "corruption" problem. Reading back thorough it now, 

Reading back through it now,

> My Search results pane has had Word wrap for years, but yesterday I lost it (pane disappeared, so, using one of the advices in [[Notepad++ Search Window disappeared](https://stackoverflow.com/questions/46632070/notepad-search-window-disappeared)), I went to "C:\Users...\AppData\Roaming\Notepad++" and renamed "config.xml" as "config.xml.BAD", which did retrieved the SR pane

I still interpret that as some sort of Notepad++ error or config file corruption that made the search result panel not appear or otherwise not function as expected, in which case there is nothing to right-click on, indicating a different, deeper problem, one that alippant/dismissive RTFM does not address.  Further, this does not indicate to me that the user simply doesn't know how to do a search to make a new search result pane open, but rather that doing sbsequent searches wasn't opening any search result pane.  But if that were the case, then word wrapping would be of least concern.  So there may be missing information, or I misinterpreted, but I rather try to understand than jump to conclusions and dismiss.  Something happened to a user that was unexpected or confusing, and it brought them here.  I'd opt to respect that experience, understand what went wrong, improve code, user interface, documentation, support process, etc.  For each user that reports a problem, there may be tens of thousands who say nothing and just stop using the software.  So there is value in feedback.  Even if the end-result was user-error, or a root problem of a different nature, little is served by insulting them or their level of proficiency.

**Comment by alankilborn** [comment]
@Leif-W 

What you say is true.
My original reply was intended for _after_ the user was past any "corruption" problem, because, as you say:

>But if that were the case, then word wrapping would be of least concern

And in regards to:

>little is served by insulting them or their level of proficiency

Reading back, I don't see an insult, but perhaps a factual comment on proficiency level.

>Something happened to a user that was unexpected

My intent was to leave _that_ part for others to comment on, but really, a "corruption" issue is best served by opening a new issue directly talking to that point, not adding it to the comment thread of some other issue.  Again, hindsight is 20/20, and probably the best comment I could have made would have been to direct them to do just that.

Oh, well, it is in the rearview mirror now, time to move on.
